src/OrchardCore/OrchardCore.ContentManagement.Display/Razor/CssOrchardHelperExtensions.cs

```diff
@@ -51,11 +58,38 @@ public static string GetFieldWrapperCssClasses(this IOrchardHelper helper, Conte
             }
         }
 
-        if (additionalClasses?.Length > 0)
+        if (additionalClasses != null)
         {
             items.AddRange(additionalClasses);
         }
 
-        return helper.GetWrapperCssClasses(items.ToArray());
+        return GetHtmlContentBuilder(items);
+    }
+
+    private static HtmlContentBuilder GetHtmlContentBuilder(IEnumerable<string> values)
+    {
+        var builder = new HtmlContentBuilder();
+
+        foreach (var value in values)
+        {
+            if (builder.Count == 0)
+            {
+                // The 'Append' method performs a string.IsNullOrEmpty() check. So, as long as the builder has no entries, we append with no spaces.
+                builder.Append(value);
+
+                continue;
+            }
+
+            if (string.IsNullOrWhiteSpace(value))
+            {
+                continue;
+            }
+
+            // At this point, we already know that the builder has at least one entry, so we append a single space to the class name.
+            // We pass create HtmlString here to prevent the builder from preforming string.IsNullOrWhiteSpace again for performance reason.
+            builder.AppendHtml(new HtmlString(' ' + value));
```

Comment:
Should be two calls to `AppendHtml`, would prevent the new string allocation (concatenation). The `HtmlString` space can be static. The value needs to be added with `Append(string)` or some user entered information (content type) could leak unenccoded.

Reply:
@sebastienros we could do this

```
builder.AppendHtml(new HtmlString(' '))); // this would be static yes
builder.AppendHtml(new HtmlString(value));
```

The reason I do no want to call `Append(value)` is because it performs `string.IsNullOrEmpty` which I was trying to avoid since we already know it contains a value.

Reply:
`Append` will encode, and encoding is required here.

Reply:
@sebastienros append does not encode. it just checks if the value is not empty.

```
    public IHtmlContentBuilder Append(string? unencoded)
    {
        if (!string.IsNullOrEmpty(unencoded))
        {
            Entries.Add(unencoded);
        }

        return this;
    }
```

the `WriteTo` method handles the encoding

Reply:
When the builder is written, anything that was added with Append will be encoded.

https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.html.htmlcontentbuilder.append?view=aspnetcore-7.0#microsoft-aspnetcore-html-htmlcontentbuilder-append(system-string)
```
Appends a [String](https://learn.microsoft.com/en-us/dotnet/api/system.string) value. The value is treated as unencoded as-provided, and will be HTML encoded before writing to output.
```

Reply:
@sebastienros I think both Append and AppendHtml will be encoded during writing. Here is the `WriteTo` implementations. 

```
    public void WriteTo(TextWriter writer, HtmlEncoder encoder)
    {
        ArgumentNullException.ThrowIfNull(writer);
        ArgumentNullException.ThrowIfNull(encoder);

        var count = Entries.Count;
        for (var i = 0; i < count; i++)
        {
            var entry = Entries[i];

            var entryAsString = entry as string;
            if (entryAsString != null)
            {
                encoder.Encode(writer, entryAsString);
            }
            else
            {
                // Only string, IHtmlContent values can be added to the buffer.
                ((IHtmlContent)entry).WriteTo(writer, encoder);
            }
        }
    }
```

Reply:
No, whatever you add as HtmlString won't.

Reply:
okay. maybe you are write. The `WriteTo` on the `HtmlString` does not make use of the encoder. Strange.

```
public class HtmlString : IHtmlContent
{
    /// <summary>
    /// An <see cref="HtmlString"/> instance for <see cref="Environment.NewLine"/>.
    /// </summary>
    public static readonly HtmlString NewLine = new HtmlString(Environment.NewLine);

    /// <summary>
    /// An <see cref="HtmlString"/> instance for <see cref="string.Empty"/>.
    /// </summary>
    public static readonly HtmlString Empty = new HtmlString(string.Empty);

    /// <summary>
    /// Creates a new <see cref="HtmlString"/>.
    /// </summary>
    /// <param name="value">The HTML encoded value.</param>
    public HtmlString(string? value)
    {
        Value = value;
    }

    /// <summary>
    /// Gets the HTML encoded value.
    /// </summary>
    public string? Value { get; }

    /// <inheritdoc />
    public void WriteTo(TextWriter writer, HtmlEncoder encoder)
    {
        ArgumentNullException.ThrowIfNull(writer);
        ArgumentNullException.ThrowIfNull(encoder);

        writer.Write(Value);
    }

    /// <inheritdoc />
    public override string ToString()
    {
        return Value ?? string.Empty;
    }
}
```

Reply:
> Strange.

No, by design. This is exactly how `HtmlContentBuilder` distinguishes between fragments it needs to encode or not.

Reply:
good to know. I updated the code let me see if you spot something else.

Reply:
Any `IHtmlContent` are intended to be already encoded

Reply:
@sebastienros I am using AppendHtml for the single space separator to avoid the need for the extra string.IsNullOrEmpry for the space. I don't think we need to worry about encoding the single space.